### PR TITLE
Move test related classes to testing-module

### PR DESCRIPTION
**Pull request by MichaelEvans**
As mentioned in #44, move some of the copy+pasted testing bits to a reusable module.

### CODE REVIEW

**Comment by JakeWharton** [comment]
Build failed. Otherwise looks good

**Comment by MichaelEvans** [comment]
Any idea why/where this comes from: 

```
/home/travis/.gradle/caches/modules-2/files-2.1/junit/junit/4.12/2973d150c0dc1fefe998f834810d68f278ea58ec/junit-4.12.jar: Error: Invalid package reference in library; not included in Android: java.lang.management. Referenced from org.junit.internal.runners.statements.FailOnTimeout. [InvalidPackage]
```

Everything I'm seeing from a Google search for this error seems to just say ignore it? Or is there a real fix?

**Comment by JakeWharton** [comment]
Is that lint?

**Comment by MichaelEvans** [comment]
Yep. Should I just configure lint to ignore that particular error type?

**Comment by JakeWharton** [comment]
Yeah

File: testing-utils/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.jakewharton.rxbinding.testingutils">
+
+  <uses-permission android:name="android.permission.DISABLE_KEYGUARD"/>
+  <uses-permission android:name="android.permission.WAKE_LOCK"/>
```
Comment (JakeWharton) [inline testing-utils/src/main/AndroidManifest.xml]: Clever putting these here

File: testing-utils/build.gradle [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  lintOptions {
+    textReport true
+    textOutput 'stdout'
+    disable 'InvalidPackage'
+  }
+
+  buildTypes {
+    debug {
+      testCoverageEnabled true
```
Comment (JakeWharton) [inline testing-utils/build.gradle]: This module has no tests!

File: testing-utils/build.gradle [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  }
+
+  dexOptions {
+    preDexLibraries = !rootProject.ext.ci
+  }
+
+  lintOptions {
+    textReport true
+    textOutput 'stdout'
+    disable 'InvalidPackage'
```
Comment (JakeWharton) [inline testing-utils/build.gradle]: They must have special-cased JUnit when it's a non-`compile` dependency.

File: testing-utils/build.gradle [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  dependencies {
+    classpath rootProject.ext.androidPlugin
+  }
+}
+
+apply plugin: 'com.android.library'
+
+dependencies {
+  compile rootProject.ext.supportTestRunner
+  compile rootProject.ext.supportTestRules
```
Comment (JakeWharton) [inline testing-utils/build.gradle]: I think this dependency and the two below it can be dropped.

**Comment by JakeWharton** [comment]
Sorry I was reviewing on my phone yesterday. Finally got a desktop-based review.

File: testing-utils/build.gradle (hunk 4 above)
Comment (MichaelEvans) [inline testing-utils/build.gradle]: It needs to depend on RxAndroid (or at least RxJava?) for the RecordingObserver.

**Comment by MichaelEvans** [comment]
No worries, there wasn't a rush to get this in obviously. Thanks for the feedback!

**Pull request merged**